app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationManager.kt

```diff
@@ -0,0 +1,100 @@
+package com.randomappsinc.studentpicker.presentation
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.content.res.AssetFileDescriptor
+import android.media.AudioManager
+import android.media.MediaPlayer
+import android.os.Handler
+import android.view.View
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.DecelerateInterpolator
+import com.randomappsinc.studentpicker.R
+import com.randomappsinc.studentpicker.utils.UIUtils
+
+class PresentationManager(
+        private val listener: Listener,
+        private val view: View) {
+
+    interface Listener {
+        fun speakNames()
+    }
+
+    private var player: MediaPlayer? = null
+    private var handler: Handler? = null
+    private val animateNamesTask = ::animateNames
+    private var defaultAnimationLengthMs: Long = 0
+
+    init {
+        handler = Handler()
+        player = MediaPlayer()
+        defaultAnimationLengthMs = view.context.resources.getInteger(R.integer.default_anim_length).toLong()
+    }
+
+    fun playSoundOrAnimations() {
```

Comment:
playSoundOrAnimations() reveals too much about the implementation, which isn't necessary. Also, if we change the nature of the presentation, we will have to rename this method. To keep it "magic"-sounding, we should just follow the name of the class. I think presentChosenName() is a fitting method name for a PresentationManager class.